libs/shared/util-types/src/lib/shared-util-types.ts

```diff
@@ -0,0 +1,18 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+// From https://stackoverflow.com/questions/52237190/infer-union-types-of-type-guards-in-typescript/52237897#52237897
+type GenericTypeGuard = (x: any) => x is any
+type UnionTypeGuard<T> = T extends (o: any) => o is infer U ? U : never
+
+export function unionTypeGuard<T extends GenericTypeGuard>(guards: T[]): (x: any) => x is UnionTypeGuard<T> {
+  return (value): value is UnionTypeGuard<T> => guards.some((guard) => guard(value))
+}
+
+type UnionToIntersection<U> = (U extends any ? (k: U) => void : never) extends (k: infer I) => void ? I : never
+type IntersectionTypeGuard<T> = UnionToIntersection<UnionTypeGuard<T>>
+
+export function intersectionTypeGuard<T extends GenericTypeGuard>(
```

Comment:
You don't use it ?

Reply:
Not for now but this can be useful in the future